src/util/triplea/MapXMLCreator/CanalDefinitionsPanel.java

```diff
@@ -0,0 +1,442 @@
+package util.triplea.MapXMLCreator;
+
+import java.awt.Color;
+import java.awt.FontMetrics;
+import java.awt.Graphics;
+import java.awt.Graphics2D;
+import java.awt.Point;
+import java.awt.event.ActionEvent;
+import java.awt.event.MouseEvent;
+import java.awt.geom.Rectangle2D;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.LinkedHashSet;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.SortedSet;
+import java.util.TreeSet;
+
+import javax.swing.AbstractAction;
+import javax.swing.JOptionPane;
+import javax.swing.JPanel;
+import javax.swing.SwingUtilities;
+
+import games.strategy.util.Tuple;
+import util.triplea.MapXMLCreator.TerritoryDefinitionDialog.DEFINITION;
+
+
+public class CanalDefinitionsPanel extends ImageScrollPanePanel {
+  private static final String NEW_CANAL_OPTION = "<new Canal>";
+
+  private SortedSet<String> selectedLandTerritories = new TreeSet<String>();
+  private SortedSet<String> selectedWaterTerritories = new TreeSet<String>();
+  private String currentCanalName = null;
+
+  private CanalDefinitionsPanel() {}

```

Comment:
The class should be marked as final as well to truly disallow just anyone creating instances, you could get more instances or override this class by inheriting from it.